docs/reference/contrib/settings.md

```diff
@@ -0,0 +1,379 @@
+# Settings
+
+The `settings` module allows you to:
+
+* Create settings that are generic across all sites
+* Create settings that are specific for a single site
+
+Settings are editable by administrators within the Wagtail admin, and can be
+accessed in code as well as in templates.
+
+## Installation
+
+Add `wagtail.contrib.settings` to your `INSTALLED_APPS`:
+
+```python
+INSTALLED_APPS += [
+    'wagtail.contrib.settings',
+]
+```
+
+**Note:** If you are using `settings` within templates, you will also need to
+update your `TEMPLATES` settings (discussed later in this page).
+
+## Defining settings
+
+Create a model that inherits from either:
+
+* `BaseGenericSetting` (for generic settings across all sites)
+* `BaseSiteSetting` (for site-specific settings)
+
+and register it using the `register_setting` decorator:
+
+```python
+from django.db import models
+from wagtail.contrib.settings.models import (
+    BaseGenericSetting,
+    BaseSiteSetting,
+    register_setting,
+)
+
+@register_setting
+class GenericSocialMediaSettings(BaseGenericSetting):
+    facebook = models.URLField()
+
+@register_setting
+class SiteSpecificSocialMediaSettings(BaseSiteSetting):
+    facebook = models.URLField()
+```
+
+Links to your settings will appear in the Wagtail admin 'Settings' menu.
+
+## Edit handlers
+
+Settings use edit handlers much like the rest of Wagtail.
+Add a `panels` setting to your model defining all the edit handlers required:
+
+```python
+@register_setting
+class GenericImportantPages(BaseGenericSetting):
```

Comment:
Generally, I don't think users should/would be prefixing these models with ``Generic``/``SiteSpecific``.

To make this slightly more realistic (and avoid establishing an unnecessary naming convention!), could we use two separate examples, one for each type?

For example: ``SocialMediaSettings(BaseGenericSetting)`` and ``ImportantPages(BaseSiteSetting)``

Reply:
I'm not sure there is a reliable real-world example we can include here that always makes sense. For instance, I need both generic and site-specific social media settings on a project I'm working on now. 

Let's be as explicitly clear as possible within the docs, and then it's up to the developer to decide on the model naming for their project.